### ISSUE TITLE
tail_n_mail hangs on super-long log line

### ISSUE THREAD

**Issue by dtseiler** [bucardo/tail_n_mail#28]
Yesterday I received a high load alert from one of my prod DB servers. I logged in to find over 10 tail_n_mail processes running.

One of my devs sent a query with a super long IN list, the log line was 6,893,029 chars long. His connection was lost (FATAL) so I assume tail_n_mail was trying to grab the statement for the email notification and choked itself into a spin. tail_n_mail was then kicking off every 5 minutes via crontab and each subsequent process was choking in the same place.

I don't think tail_n_mail needs to print the entire statement in an email, and really I wouldn't want to print this one and eat up my inbox space. Not sure if you can grab just a subset of 1,000 or 10,000 chars without having to read the whole thing.

**Comment by dtseiler** [comment]
I have the minimal snippet of the log lines (6 lines) to duplicate the error in tail_n_mail. However even gzipped the file is 7.3M. Do you want that attached here? I'm guessing you could create your own such lines with a generated query with a long IN list.

**Comment by dtseiler** [comment]
FYI this system is CentOS 7.4, perl 5.16.3, tail_n_mail 1.31.3

**Comment by turnstep** [comment]
Give the "truncate_very_large_lines" branch a try. It truncates the lines at 10,000 characters by default, just before we start all the heavy regexes and string manipulations. You can adjust it via --read-line-max-size=XXX. Setting it to 100,000 is a noticeable difference in speed, but 10k seems a pretty decent default as savings are minimal going below that. However, I've only done minimal mock testing: please give this a run on your real giant log files and let me know if there is a better default than 10k.

**Comment by darshansh** [comment]
Hello, 
I tried to configured tail_n_mail on my windows machine. But not getting mail. Can anybody please share steps for same ? 

![image](https://user-images.githubusercontent.com/14271741/55231214-f484a480-5247-11e9-8d0d-2534a7d1fc87.png)

**Comment by turnstep** [comment]
@darshansh Please raise a separate issue, or use the mailing list. Looks like you need to provide a mailserver.

@dtseiler Can we close this issue?

**Comment by turnstep** [comment]
Closing as hopefully solved. :)
